browse/routes/ui.py

```diff
@@ -30,21 +31,43 @@ def apply_response_headers(response: Response) -> Response:
 
 
 @blueprint.route('/abs', methods=['GET'])
-def bare_abs() -> None:
-    """Return 404."""
-    raise NotFound
+def bare_abs() -> Response:
+    """Check several legacy request parameters."""
+    if request.args:
+        if 'id' in request.args:
+            return abstract(request.args['id'])
+        elif 'archive' in request.args and 'papernum' in request.args:
+            return abstract(
+                f"{request.args['archive']}/{request.args['papernum']}")
+        else:
+            for param in request.args:
+                # singleton case, where the parameter is the value
+                # e.g. /abs?<archive>/\d{7}
+                if not request.args[param] \
+                   and re.match(r'^[a-z\-]+(\.[A-Z]{2})?\/\d{7}$', param):
+                    return abstract(param)
+
+    """Return abs-specific 404."""
+    raise AbsNotFound
 
 
 @blueprint.route('/abs/', methods=['GET'], defaults={'arxiv_id': ''})
-@blueprint.route('/abs/<path:arxiv_id>', methods=['GET', 'POST'])
-def abstract(arxiv_id: str) -> Union[str, Response]:
+@blueprint.route('/abs/<path:arxiv_id>', methods=['GET'])
+def abstract(arxiv_id: str) -> Response:
```

Comment:
I was wondering why it had been `Union[str, Response]` in the past... this makes more sense, I think. Also glad that using the `abstract` function to service multiple controllers worked out.